### PR TITLE
Add introduction to MeiliSearch (#484)

### LINKED ISSUE #484: Introduction to MeiliSearch
We should add a guide/introduction where we explain very concisely for what meilisearch is used for.
The following information should be added:
- Not your main database
- Not made for huge full-text datasets
- Made for site-search
- made to be fast, very fast
- made for instant search (front to meilisearch and not front to back to meilisearch)

### PR DESCRIPTION
**Pull request by tpayet**
closes #484

### CODE REVIEW

File: guides/README.md [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+
+MeiliSearch aims to be your go-to search backend when you want to build a great search experience for your end-users.
+MeiliSearch is designed for end-user search in a not-so-big data collection (< 10M documents).
+MeiliSearch is made for [type-as-you-search](https://en.wikipedia.org/wiki/Incremental_search) and [prefix-search](https://en.wikipedia.org/wiki/Trie).
+
+## ⛔️ Anti-pattern
+
+MeiliSearch should not be your main data store. MeiliSearch should contain only the data you want your users to search through. The more data MeiliSearch contains, the less relevant it is.
+
+MeiliSearch queries should be sent directly from the interface. The more proxy there is between MeiliSearch and the end-user, the less fast queries and thus search-eperience will be.
```
Comment (curquiza) [inline guides/README.md:13]:
```suggestion
MeiliSearch queries should be sent directly from the interface. The more proxy there is between MeiliSearch and the end-user, the less fast queries and thus search-experience will be.
```

[the remark this answers is not in the document]
Comment (curquiza) [reply guides/README.md:13]: I was a little bit confused with "interface" too. For me, "from the interface" would mean "from the MeiliSearch interface" in this context. Not the "user front-end interface". But maybe I'm the only one.

[the remark this answers is not in the document]
Comment (tpayet) [reply guides/README.md]: I went with front-end since it's more clear

File: guides/README.md [hunk 2]
```diff
@@ -7,3 +21,7 @@ Content:
 - [🚀 Introduction](/guides/introduction/): Starting with MeiliSearch!
 - [💡 Main Concepts](/guides/main_concepts/): Understanding the basics like indexes, documents and searches.
 - [📚 Advanced Guides](/guides/advanced_guides/): Deep diving into the advanced but accessible concepts of MeiliSearch.
+
+::: note
+This [documentation is completely open-source](https://github.com/meilisearch/documentation). We keep it up-to-date but you might find some typos or mistakes. Help us make it right by submitting a pull request or an issue 😁
```
Comment (matboivin) [inline guides/README.md:26]:
```suggestion
This [documentation is completely open-source](https://github.com/meilisearch/documentation). We keep it up-to-date but you might find some typos or mistakes. Help us make it a better guide by submitting a pull request or an issue 😁
```

File: guides/README.md [hunk 3]
```diff
@@ -7,3 +21,7 @@ Content:
 - [🚀 Introduction](/guides/introduction/): Starting with MeiliSearch!
 - [💡 Main Concepts](/guides/main_concepts/): Understanding the basics like indexes, documents and searches.
 - [📚 Advanced Guides](/guides/advanced_guides/): Deep diving into the advanced but accessible concepts of MeiliSearch.
+
+::: note
+This [documentation is completely open-source](https://github.com/meilisearch/documentation). We keep it up-to-date but you might find some typos or mistakes. Help us make it a better guide by submitting a pull request or an issue 😁
```
Comment (curquiza) [inline guides/README.md:26]:
```suggestion
This [documentation is completely open-source](https://github.com/meilisearch/documentation). We keep it up-to-date but you might find some typos or mistakes. Help us make it a better guide by submitting a pull request or [an issue](https://github.com/meilisearch/documentation/issues) 😁
```

**Pull request merged**